frontend/app/src/App.tsx

```diff
@@ -925,6 +925,9 @@ export class App extends PureComponent<Props, State> {
           appPages: newSessionProto.appPages,
           currentPageScriptHash: newPageScriptHash,
           latestRunTime: performance.now(),
+          // If we're here, the fragmentIdsThisRun variable is always the
+          // empty array.
+          fragmentIdsThisRun,
```

Comment:
I couldn't think of a good way to write a unit test for this one since RTL doesn't allow you to write tests against the internal state of a component 😞 

Suggestions for how to do this would be appreciated if anyone has any ideas!

Reply:
yeah, a unit test for this looks complicated. Can we somehow test that the mentioned issue doesn't appear with an e2e test?

Reply:
Hm, an e2e test is unfortunately pretty difficult to do given the way that the bug manifests is probabilistically (and also pretty rarely 😐). I'll try thinking of something while I'm working on more complete e2e tests for the feature